### ISSUE TITLE
Minification issue

### ISSUE THREAD

**Issue by peterflex7** [scandipwa/base-theme#503]
I have just install Scandipwa-installer & generate theme and run npm ci && npm run build on my server  but i have a blank page on my domain with some errors on console. 
did i have miss something ?

server Nginx
varnish & redis setup properly.

<img width="940" alt="Screenshot 2020-01-21 at 5 05 01 PM" src="https://user-images.githubusercontent.com/26825774/72803742-9dd1d680-3c70-11ea-8c60-66275a797644.png">

**Comment by alfredsgenkins** [comment]
Well, the page shows 404 for those resources. Are they there? Can you validate app/design/frontend/Scandiweb/pwa/Magento_Theme is present on a server, has web folder inside with all  (most neccessay) assets - language bundles, css, etc?

**Comment by alfredsgenkins** [comment]
Does Magento 2 recognized a theme? Can you see it in BE or database?

**Comment by peterflex7** [comment]
Yes its recognized the theme & also 
<img width="678" alt="Screenshot 2020-01-21 at 5 29 41 PM" src="https://user-images.githubusercontent.com/26825774/72805049-c4454100-3c73-11ea-9396-776b6a6d40d2.png">

**Comment by peterflex7** [comment]
<img width="1398" alt="Screenshot 2020-01-21 at 5 33 40 PM" src="https://user-images.githubusercontent.com/26825774/72805229-3027a980-3c74-11ea-83e6-57b04025dc68.png">

**Comment by alfredsgenkins** [comment]
You are using / as root, must be /pub

**Comment by alfredsgenkins** [comment]
Please see the themes readme. You can edit this path if really needed in webpack production config - public path.

**Comment by alfredsgenkins** [comment]
But please prefer public path in pub, not in the project root :)

**Comment by peterflex7** [comment]
const publicPath = `/pub/static/version${staticVersion}/frontend/Scandiweb/pwa/en_US/Magento_Theme/`;

i have configured this but still the same

**Comment by peterflex7** [comment]
<img width="617" alt="Screenshot 2020-01-21 at 6 06 14 PM" src="https://user-images.githubusercontent.com/26825774/72807223-ba720c80-3c78-11ea-89d1-06cc4c7e48a6.png">

**Comment by alfredsgenkins** [comment]
OK, seems like CSS has been generated. Can you please share the content of `/assets/` folder?

**Comment by peterflex7** [comment]
<img width="669" alt="Screenshot 2020-01-21 at 6 16 19 PM" src="https://user-images.githubusercontent.com/26825774/72807888-230db900-3c7a-11ea-925a-20728f14e3ac.png">

**Comment by alfredsgenkins** [comment]
Well, it seems like an issue with Html component - this code is for some reason compiled in a wrong way:

```
replaceLinks({ attribs, children }) {
        const { href, ...attrs } = attribs;

        if (href) {
            const isAbsoluteUrl = value => new RegExp('^(?:[a-z]+:)?//', 'i').test(value);
            const isSpecialLink = value => new RegExp('^(sms|tel|mailto):', 'i').test(value);

            if (!isAbsoluteUrl(href) && !isSpecialLink(href)) {
                return (
                    <Link { ...attributesToProps({ ...attrs, to: href }) }>
                        { domToReact(children, this.parserOptions) }
                    </Link>
                );
            }
        }
    }
```

May I ask you to give me more details on base-theme (`scandipwa/source`) version? Please use composer for that. Thanks!

I am thinking this could be an issue with our dependencies, can you validate your package.json and lock is the same as in `vendor/scandipwa/source`?

Otherwise, I could advice you rewriting an Html to fix an issue.

We never experienced this issue before.

**Comment by alfredsgenkins** [comment]
The code is here for 5 month - strange. Can I also ask you node version?

**Comment by peterflex7** [comment]
[master_q]:~$ node -v
v12.1.0
[master_q]:~$ npm -v
6.9.0

  <module name="ScandiPWA_Source" setup_version="0.0.1" >

**Comment by peterflex7** [comment]
<module name="ScandiPWA_Source" setup_version="0.0.1" >

**Comment by alfredsgenkins** [comment]
No, I mean: `composer info scandipwa/source` :)

**Comment by peterflex7** [comment]
name     : scandipwa/source
descrip. : ScandiPWA
keywords : magento2, pwa, scandipwa, source
versions : * 2.0.1
type     : magento2-theme
license  : Open Software License 3.0 (OSL-3.0) (OSI approved) https://spdx.org/licenses/OSL-3.0.html#licenseText
source   : [git] https://github.com/scandipwa/base-theme.git 841f54f12ab7d4ab64316ec5ff30b458a967ef40
dist     : [zip] https://api.github.com/repos/scandipwa/base-theme/zipball/841f54f12ab7d4ab64316ec5ff30b458a967ef40 841f54f12ab7d4ab64316ec5ff30b458a967ef40
path     : /home/350054.cloudwaysapps.com/vqtfvngwer/public_html/vendor/scandipwa/source
names    : scandipwa/source

autoload
files

requires
magento/magento2-base 2.3.1
scandipwa/catalog-graphql ^1.3
scandipwa/cms-graphql ^1.0
scandipwa/customer-graph-ql ^1.1
scandipwa/menu-organizer ^1.0
scandipwa/persisted-query ^1.0
scandipwa/quote-graphql ^2.0.1
scandipwa/reviews-graphql ^1.0
scandipwa/route717 ^1.0
scandipwa/slider-graphql ^1.0
scandipwa/urlrewrite-graphql ^1.0
scandipwa/wishlist-graphql ^1.0

**Comment by alfredsgenkins** [comment]
This version is hyper old. Is this intended? Can you please run composer update `scandipwa/*`?

**Comment by peterflex7** [comment]
i have run composer update scandipwa/source 
its return nothing to update

**Comment by alfredsgenkins** [comment]
Can you please `composer why-not scandipwa/source:2.8.2`?

**Comment by peterflex7** [comment]
its seems like i have to update my magento to 2.3.3 first :)

**Comment by alfredsgenkins** [comment]
Okey - I see only one option - the node version. ScandiPWA is built using 10.x version. Can you try downgrading?

Seems popular on the web:

```Its compatibility issue of Nodejs I uninstalled my Node(12.11) and installed Node(10) stable version and it worked perfectly.```

**Comment by peterflex7** [comment]
but on my local environment i running scandi on lastest node version ? but on my server i'm getting error is their anything i'm missing on configuration ?
i have just install scandi installer node and npm 
configure redis and varnish 

root@f9c48d671c21:/var/www/public# node -v
v12.14.0
root@f9c48d671c21:/var/www/public#

**Comment by alfredsgenkins** [comment]
Are you running it using frontend (webpack-dev-server) on local, or via the compiling and later serving by Magento - this makes a big difference as most "uglifying" plugins do not work in that case.

**Comment by alfredsgenkins** [comment]
This is true:
- i have just install scandi installer node and npm
- configure redis and varnish

The issue you are experiencing is a bug, it should not be like this. I can not confirm if it is a node problem, or what - but if it is, we would know how to deal with it in the future + will plan a fix for it.

**Comment by peterflex7** [comment]
i'm using docker-compose on local @alfredsgenkins its working fine on local 
but on live environment(using composer install) by following this 
Magento
These changes are theme installation specific and couple things are required

Theme install
The ScandiPWA theme is distributed through scandipwa/installer To install theme on running Magento you need to execute:

composer require scandipwa/installer to fetch and install PWA components
magento scandipwa:theme:bootstrap Scandiweb/pwa to install theme files from vendor to proper place in Magento themes
magento setup:upgrade to register PWA theme in magento
Build theme, for example, like this
cd $BASEPATH/app/design/frontend/Scandiweb/pwa || exit
npm ci
npm run build
Copy to clipboardErrorCopied
Persisted Query
The persisted queries are cached in Redis, each time graphql receives a query, its response is cached with unique hash in redis, if reposnse changes, new hash will be used.

Magento configuration must be set in order to use Redis as storage for persisted queries

bin/magento setup:config:set \
    --pq-host=redis \
    --pq-port=6379 \
    --pq-database=5 \
    --pq-scheme=tcp

**Comment by alfredsgenkins** [comment]
But if you are using our docker-compose then it runs the project in `10.x` node :)

**Comment by alfredsgenkins** [comment]
You step-by-step seems exactly right!

**Comment by peterflex7** [comment]
let me try again after downgrading the node to v10 stable

**Comment by alfredsgenkins** [comment]
Thank you sir!

**Comment by peterflex7** [comment]
Thanks @alfredsgenkins appreciated for help i will get back to you after testing this :)

**Comment by peterflex7** [comment]
@alfredsgenkins  i just downgrade to node 10.x and tried but i'm getting same errors with blank frontend :(

**Comment by peterflex7** [comment]
@alfredsgenkins i just find out the issue & solution :)

**Comment by alfredsgenkins** [comment]
@peterflex7 Could you please share - so we can mark this issue as resolved? :)

**Comment by peterflex7** [comment]
That issue was due to Minify JavaScript Files i just turn that off now its working but their is a grapql error cannot fetch data now
<img width="1211" alt="Screenshot 2020-01-22 at 4 38 06 PM" src="https://user-images.githubusercontent.com/26825774/72891338-d8ea0d80-3d35-11ea-9be3-c64125dd6052.png">

**Comment by alfredsgenkins** [comment]
Looks like something is wrong with your `createEmptyCart` query. Which causes all other issues. I am unsure what could this be :( Any logs on server side?

**Comment by alfredsgenkins** [comment]
Minification issue - how strange, but thanks it is noted.

**Comment by peterflex7** [comment]
Thanks @alfredsgenkins its also resolved its due to graphql i just enable default payment extension i.e amazon etc graphql issue resolved now the application working properly :) Thanks for Help once again

**Comment by peterflex7** [comment]
@alfredsgenkins any idea ?
{"errors":[{"message":"Variable \"$cartData_guestCartId_0\" got invalid value {\"type\":\"SHOW_NOTIFICATION\",\"msgType\":\"error\",\"msgText\":\"Internal server error\"}; Expected type String; String cannot represent an array value: {\"type\":\"SHOW_NOTIFICATION\",\"msgType\":\"error\",\"msgText\":\"Internal server error\"}","category":"graphql"}]}
